### ISSUE TITLE
MUSET fails on initial test data

### ISSUE THREAD

**Issue by valzip** [CamilaDuitama/muset#19]
Hi Camila!
I tried to run MUSET test data from github at out server, but I am getting error and abundance matrix is not being generated.
Our system:
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy
$ uname -rvp
5.15.0-113-generic #123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024 x86_64
```
I have installed MUSET from bioconda:
```
$ conda list
# packages in environment at /data/users/zeman/miniconda3/envs/muset_env:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
ggcat                     1.1.0                hc9368f3_0    bioconda
libgcc                    14.2.0               h77fa898_1    conda-forge
libgomp                   14.2.0               h77fa898_1    conda-forge
libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge
libzlib                   1.3.1                hb9d3cd8_2    conda-forge
muset                     0.5                  h6ad138b_1    camiladuitama
zlib                      1.3.1                hb9d3cd8_2    conda-forge
```
In the test folder:
```
$ muset --file fof.txt
[2024-12-11 14:25:10] [info] Running muset v0.5
[2024-12-11 14:25:10] [info] ---------------
[2024-12-11 14:25:10] [info] input file (--file): fof.txt
[2024-12-11 14:25:10] [info] output directory (-o): output
[2024-12-11 14:25:10] [info] k-mer size (-k): 31
[2024-12-11 14:25:10] [info] minimum abundance (-a): 2
[2024-12-11 14:25:10] [info] minimum unitig length (-l): 61
[2024-12-11 14:25:10] [info] minimum unitig fraction (-r): 0
[2024-12-11 14:25:10] [info] write unitig sequence (-s): false
[2024-12-11 14:25:10] [info] output unitig fraction matrix (--out-frac): false
[2024-12-11 14:25:10] [info] input consists of logan unitigs (--logan): false
[2024-12-11 14:25:10] [info] minimizer size (-m): 15
[2024-12-11 14:25:10] [info] fraction of absent samples (-f): 0.1
[2024-12-11 14:25:10] [info] fraction of present samples (-F): 0.1
[2024-12-11 14:25:10] [info] keep temporary files (--keep-temp): false
[2024-12-11 14:25:10] [info] threads (-t): 4
[2024-12-11 14:25:10] [info] ---------------
[2024-12-11 14:25:10] [info] Building k-mer matrix with kmtricks
[2024-12-11 14:25:10] [info] Run with Kmer<32> - uint64_t implementation
ggcat_cmdline 0.1.0
[2024-12-11 14:25:10] [info] Compute configuration...
[2024-12-11 14:25:10] [info] 10 samples found (20 read files).
[2024-12-11 14:25:10] [info] Use 4 partitions.
[2024-12-11 14:25:10] [info] Compute minimizer repartition...
Compute SuperK   [==================================================] [00m:01s]                                                             
Count partitions [==================================================] [00m:02s]                                                             
Merge partitions [==================================================] [00m:00s]                                                             
[2024-12-11 14:25:12] [info] Filtering k-mer matrix
[2024-12-11 14:25:12] [info] filtering kmtricks matrix: output/kmer_matrix
[2024-12-11 14:25:12] [info] 5097/5097 kmers retained
[2024-12-11 14:25:12] [info] Writing k-mers in FASTA format
[2024-12-11 14:25:12] [info] 5097 k-mers processed
[2024-12-11 14:25:12] [info] Building unitigs
[2024-12-11 14:25:12] [info] Filtering unitigs
[2024-12-11 14:25:12] [error] input file output/unitigs does not exist
(muset_env) zeman@HPC:~/ANALYSIS/muset/test$ Allocator initialized: mem: 2 GiB chunks: 8192 log2: 18
Using m: 12 with k: 31
Started phase: reads bucketing
Temp buckets files size: 2.10 MiB
Finished phase: reads bucketing. phase duration: 289.54ms gtime: 289.55ms
Started phase: kmers merge
Finished phase: kmers merge. phase duration: 542.15ms gtime: 831.72ms
Started phase: hashes sorting
Finished phase: hashes sorting. phase duration: 103.87ms gtime: 935.61ms
Started phase: links compaction
Completed compaction with 11 iters
Finished phase: links compaction. phase duration: 402.97ms gtime: 1.34s
Started phase: reads reorganization
Finished phase: reads reorganization. phase duration: 105.27ms gtime: 1.44s
Started phase: unitigs building
Finished phase: unitigs building. phase duration: 23.58ms gtime: 1.47s
Compacted De Bruijn graph construction completed.
TOTAL TIME: 1.47s
Final stats:
	phase: reads bucketing 	=> 289.54ms
	phase: kmers merge 	=> 542.15ms
	phase: hashes sorting 	=> 103.87ms
	phase: links compaction 	=> 402.97ms
	phase: reads reorganization 	=> 105.27ms
	phase: unitigs building 	=> 23.58ms
Final output saved to: output/unitigs
```
output folder contains these files:
```
$ ll output/
total 484K
drwxrwx---  4 zeman zeman 4.0K Dec 11 14:29 ./
drwxrwx---  3 zeman zeman 4.0K Dec 11 14:29 ../
-rw-rw----  1 zeman zeman    0 Dec 11 14:29 ggcat.log
drwxrwx---  2 zeman zeman 4.0K Dec 11 14:29 ggcat_build_temp/
drwxrwx--- 15 zeman zeman 4.0K Dec 11 14:29 kmer_matrix/
-rw-rw----  1 zeman zeman 189K Dec 11 14:29 matrix.filtered.fasta
-rw-rw----  1 zeman zeman 259K Dec 11 14:29 matrix.filtered.mat
-rw-rw----  1 zeman zeman 1.8K Dec 11 14:29 muset_20241211_142901.log
-rw-rw----  1 zeman zeman 5.9K Dec 11 14:29 unitigs
-rw-rw----  1 zeman zeman  648 Dec 11 14:29 unitigs.stats.log

```
unitig file is present and is not empty:
```
$ head output/unitigs
>0 LN:i:350
AAGAAGGGAGAAAGAGAGAAAGAAGAAGAGGGAAGAGGAGAAGAAGGAGGGGGGGGAGAGGGGGGGAGAGGTGAGAAAGAGACAGGGGGAAGAGAAGCAGAAGGAAGGGGGGAGAGGGGGGCGGGAGGAGAAGAGCAAGGGGGGGAGGGGCCGGGTCGGGCCGGGCCGGGCTGCGCGCACGCACCGGTCCTGGGCGCGGATCCGCCGCAGCTCATCGGCGACCCATAAACCCGTCCAACGAAGGTGGTGCTCACCCTAGCTGCGTGGACCCATCACGCGCACTCCGAGACACATACATTCCCACGCGCGTCAGGGGAAACGCGCCGTCGCGCGCGCAGTTCCTCGTGTAG
>1 LN:i:251
AAGAGGAAAAAAGAGGAAAAGAAAAGGAAGGAAGGGGGGAAAGGGAGGAAGGGTAAAAAAAGTGGTAAAAGAGGAAAAAAGAGAGAAGAAAAAAGAGAAGGAAAAGAAAACGAGAGGGAGGGGGAGAGACAGGAGAAGGAAAGAAAGGGGGGAAGGGAGAAGGAGGAAAAAGAAAGGGAGGGATAAAAAGTAATGAGGAAGGGAAGAGGTCGCAAGAAAGAGGGAAGAGAAAAAGAAGGAGAGATTAAAGG
>2 LN:i:280
GGTTGCAATCAACTTAAAGTGTTATGTGGAACATTAAAGAATATAAAGGAATCCACGTCATTAATTATTATAAAAAATATACAATGGGAAGTTGGAAGGAGAAACTAATTTATTTATTCATTCTGTTAATATCTTTACCTGTTCAATATCTCCTAATAATCTAATTAAATACAATTTTTTGTGGAGGGGCTACATCTCCCAAAGAGGCATTTTTCAGTTAGAATTCATTACTGCTTTGAAATAATAAATTTTAATTCTATGGTTATAATCTTATGTAAGA
>3 LN:i:251
CCCTCAATTTTTCCTTTTCTTTTTTCTTTTTCTTCCTTTTTTTTCTTTTCTTTTCCTCTTTCTTTCTTTTTTCCCTTTCTTCCCTTTCTTTCTCCTTCTTCTCTTTTTCTTCTCTTTTTCCTTCTTTCTTCTTCTTCCTTTTCTTCTTCTCCTCTTTCCTTTTCTCTCCGTTTCTTCCTTTTTTCGCCGCTTTCTTTTTTTTCTTCTTCTCTTTTTTGTTCCCTTTCCCTCTCTCCCCCCCTTTTTCTTTT
>4 LN:i:251
AAAAAAGGAAGGAAAAAAGGGAGAAAGTAGGAGAGGGAAGAAAGGAAAGGAAGCAGAAAAGACCAAAGAGAAGGAAGACGAAGAAAGAAGGGGAGAAAGAAAGCAGAAACAGAAGAAGAAGAAAGAACAGAAAGAAAAGGAGAGAGAAAAGAGAAAGAAAAGGGAGGAAAAAAGGAGAAAAAGAAAAAGATGGAAAAAAAAAAAAGAAGAAAGAAAAAAAAAAACAAGAAGAATGGAGAGAAAAAAAAGAG
```
Tried with -f 0 and -F 0 options but it results in the same ` [error] input file output/unitigs does not exist`

Any advices? Thanks

**Comment by rvicedomini** [comment]
Hi Michal,

Thank you for reporting us the bug you ran into.
I released a new version (0.5.1) which should fix it.

Best,
Riccardo

**Comment by valzip** [comment]
Hi Riccardo,
your fix is working. There are no troubles this time. Thanks!
Best 
Michal
